File: src/main/java/org/jetbrains/pmdkotlin/lang/kotlin/ast/AbstractKotlinNode.java
package org.jetbrains.pmdkotlin.lang.kotlin.ast;

import com.intellij.psi.PsiElement;
import net.sourceforge.pmd.lang.ast.Node;

public interface AbstractKotlinNode extends Node {
    Object jjtAccept(KotlinParserVisitor visitor, Object data);

    Object childrenAccept(KotlinParserVisitor visitor, Object data);

    PsiElement getInnerNode();
    <T> Class<T> getPsiClass();
}

File: src/main/kotlin/org/jetbrains/pmdkotlin/lang/kotlin/ast/AbstractKotlinNode.kt
package org.jetbrains.pmdkotlin.lang.kotlin.ast

import com.intellij.psi.PsiElement
import net.sourceforge.pmd.lang.ast.Node

interface AbstractKotlinNode : Node {
    fun jjtAccept(visitor: KotlinParserVisitor, data: Any): Any

    fun childrenAccept(visitor: KotlinParserVisitor, data: Any): Any

    val innerNode: PsiElement

    fun <T> getPsiClass(): Class<T>
}
